### PR TITLE
Feat/update user entity

### PR DESCRIPTION
**Pull request by klavenjones**
Made updates to  user by adding the password column of the user

- Modified User Entity
- Modified User Controller
- Modified User Service 
- Add Migration to reflect changes to the user entity

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,10 +28,10 @@ jobs:
 
     steps:
       - name: Copy repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
```
Comment (JeromeGoodrich) [inline .github/workflows/ci.yml]: could a be a separate small PR since it doesn't relate to updating the user entity.

File: src/api/user/dto/user.dto.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export class UserDto {
+  @IsString()
+  @IsNotEmpty()
+  @MaxLength(120)
+  public username: string;
+
+  @IsString()
+  @IsNotEmpty()
+  @MaxLength(60)
```
Comment (JeromeGoodrich) [inline src/api/user/dto/user.dto.ts]: Beware magic numbers 👻 ! Why is a max length of 60 important? Passwords seem important are there other validations that we might want here?

File: src/database/setup.ts [hunk 3]
```diff
@@ -24,16 +24,6 @@ const dbQuery = (query: string, successMessage: string, exitScript = false) => {
...
-//Creating a role for the user.
-const createRole = `CREATE ROLE ${process.env.DATABASE_USER} LOGIN CREATEDB`;
-dbQuery(createRole, `CREATED ROLE NAMED ${process.env.DATABASE_USER} SUCCESFULLY`);
-
-//Add password to role if provided
-if (process.env.DATABASE_PASSWORD) {
-  const addPassword = `ALTER ROLE ${process.env.DATABASE_USER} with PASSWORD ${process.env.DATABASE_PASSWORD}`;
-  dbQuery(addPassword, 'PASSWORD ADDED TO ROLE SUCCESSFULLY');
-}
-
```
Comment (JeromeGoodrich) [inline src/database/setup.ts:36]: This seems like it should belong in a different PR. It has nothing to do with updating the user entity

**Comment by JeromeGoodrich** [comment]
I'd love to see some more thought go into the password validation.

**Comment by erichuiwong** [comment]
Could you add a summary in the description of the PR on what's specifically being modified?
